File: abc159/b/b.go
package main

import "fmt"

func kaibun(s string) int {
	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
		if s[i] != s[j] {
			return 0
		}
	}
	return 1
}

func main() {
	var s string
	fmt.Scan(&s)
	judge := 0
	sl := len(s)

	judge += kaibun(s[:(sl-1)/2])
	judge += kaibun(s[(sl+3)/2-1:])
	judge += kaibun(s)

	if judge == 3 {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}

File: abc159/b/b.cpp
#include <iostream>
#include <string>

using namespace std;

int kaibun(string s, int start, int end) {
  while (start < end) {
    if (s[start] != s[end]) {
      return 0;
    }
    start++;
    end--;
  }
  return 1;
}

int main(void) {
  string s;
  cin >> s;
  int judge = 0;
  int sl = s.length();

  judge += kaibun(s, 0, (sl - 1) / 2 - 1);
  judge += kaibun(s, (sl + 3) / 2 - 1, sl - 1);
  judge += kaibun(s, 0, sl - 1);

  if (judge == 3) {
    cout << "Yes" << "\n";
  } else {
    cout << "No" << "\n";
  }
  return 0;
}
